posts/2021-07-08-sig-bugs.adoc

```diff
@@ -0,0 +1,133 @@
+image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]
+
+//tag::features[]
+
+//end::features[]
+
+[#bugs]
+== Notable bugs fixed in this release
+
+
+We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A21.0.0.7+label%3A%22release+bug%22[full list of bugs fixed in 21.0.0.7].
+
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17344[Occasional login failure of OIDC RP if clientSecret is not configured]
++
+In the `OIDC RP`, when no `clientSecret` is configured, validation of the signature of the `JWT` occasionally would fail with a message similar to:
++
+[source, text]
+----
+CWWKS1739E: A signing key required by signature algorithm [none] was not available. null
+----
++
+This bug was fixed by updating the `JWT` validation code in the `OpenID` Connect client to return a `null` `java.security.Key` object as the signature verification key if `signatureAlgorithm` is set to `none` in the `OpenID` Connect client configuration. Before, an `HmacKey` object was instantiated using the `clientSecret` value from the `OIDC` client configuration, even though the Key object will never be used if `signatureAlgorithm` is set to `none`. The new behavior avoids a `NullPointerException` when a `clientSecret` is not configured in the `OIDC` client.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17593[EJB Singleton Lifecycle Deadlock]
++
+A Singleton can be locked from the lifecycle event causing a deadlock when:
+
+** When the Singleton called another managed object (like another session bean or asynchronous bean) OR the Singleton had a lifecycle interceptor.
+** The other managed object calls the Singleton. (Either through an asynchronous method that is invoked at the right time (so a timing problem) or through the method the Singleton called on it).
+
+For `@PreDestroy` we are no longer locking the singleton, depending on when the destroy resolves, the other managed object might get an error stating the bean has been stopped/destroyed.
+
+For `@PostConstruct` the `EJBContainer` must lock the creation of `SingletonBeans` to ensure only one gets created. Instead of deadlocking forever we now have a timeout value of attempting to obtain a lock. This value is 2 minutes by default but configurable by `@AccessTimeout` value either on the class or the `@PostConstruct` method. A thread that times out trying to access the Singleton will get a `javax.ejb.ConcurrentAccessTimeoutException`.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17235[Improve featureUtility return code when invalid action name is specified]
++
+Users would see an erroneous `return code 0` when an invalid action was specified. This was now changed to `BAD_ARGUMENT (20)`. Because of this change any scripted invocations of featureUtility can now detect this.
```

Comment:
I would update this sentence to `This was now changed to  `return code 20`. and remove the BAD_ARGUMENT reference from the post. BAD_ARGUMENT is the coded name for RC=20 but we don't refer to it in the docs really so it my cause confusion leaving it in.

Reply:
Great, thank you!

Done